### PR TITLE
handle corrupted job status file

### PR DESCRIPTION
**Pull request by suwatch**
Treat corrupted job status file as initial job run state (as if no job status at all).

### CODE REVIEW

File: Kudu.Core/Jobs/JobsManagerBase.cs [hunk 1]
```diff
@@ -422,7 +422,7 @@ private static string GetJobSettingsPath(DirectoryInfoBase jobDirectory)
 
         protected TJobStatus GetStatus<TJobStatus>(string statusFilePath) where TJobStatus : class, IJobStatus, new()
         {
-            return JobLogger.ReadJobStatusFromFile<TJobStatus>(TraceFactory, statusFilePath) ?? new TJobStatus();
+            return JobLogger.ReadJobStatusFromFile<TJobStatus>(TraceFactory, statusFilePath);
```
Comment (davidebbo) [inline Kudu.Core/Jobs/JobsManagerBase.cs]: No sure I understand why it's wrong to return a `new` object when `ReadJobStatusFromFile` returns null.

Comment (davidebbo) [inline Kudu.Core/Jobs/JobsManagerBase.cs]: Ok, saw your comments in mail.

**Comment by davidebbo** [comment]
:shipit:

**Comment by suwatch** [comment]
https://github.com/projectkudu/kudu/commit/63e4c4003bf856ca23d0c319b024684d7765df87

**Pull request closed without merging**